src/server/engine_shard.cc

```diff
@@ -716,7 +732,33 @@ void EngineShard::RetireExpiredAndEvict() {
     ttl_delete_target = kTtlDeleteLimit * double(deleted) / (double(traversed) + 10);
   }
 
-  ssize_t eviction_redline = size_t(max_memory_limit * kRedLimitFactor) / shard_set->size();
+  const size_t shards_count = shard_set->size();
+  const double eviction_memory_budget_threshold = GetFlag(FLAGS_eviction_memory_budget_threshold);
+  const double rss_oom_deny_ratio = ServerState::tlocal()->rss_oom_deny_ratio;
+
+  const bool should_evict =
+      eviction_memory_budget_threshold > 0.0 && GetFlag(FLAGS_enable_heartbeat_eviction);
```

Comment:
why do you check eviction_memory_budget_threshold > 0.0 ?
nit: maybe rename this var should_evict -> eviction_enabled

Reply:
What does it mean when `eviction_memory_budget_threshold` is less than 0? I believe it indicates that this flag is disabled

Reply:
I dont think we need this logic to disable the flag as we already have enable_heartbeat_eviction flag that disables/enables the heartbeat , and this logic with eviction_memory_budget_threshold > 0 is redundant.
In general if you create such logic to a flag than document it in the flag string description